python/ray/serve/batching.py

```diff
@@ -523,3 +587,17 @@ async def batch_wrapper(*args, **kwargs):
     # with the underlying function as the sole argument (i.e., it must be a
     # "decorator factory.").
     return _batch_decorator(_func) if callable(_func) else _batch_decorator
+
+
+def _set_result_safe(future: asyncio.Future, result: Any):
+    """Sets the future's result if the future is not done."""
+
+    if not future.done():
+        future.set_result(result)
+
+
+def _set_exception_safe(future: asyncio.Future, exception: Any):
+    """Sets the future's exception if the future is not done."""
+
+    if not future.done():
+        future.set_exception(exception)
```

Comment:
It's unclear from these functions what "safe" means -- either expand in the docstring that this is "safe" because otherwise it might raise an exception or else rename to something more self-describing like "_set_future_if_not_done`

Reply:
Makes sense, I changed the function names to be self-describing.